### PR TITLE
Front flmastor

### PR DESCRIPTION
**Pull request by Mastodoonte**
**Front Pull request - Auth/Context API **

General Auth:
 _- Implementation of Context Api from root-> data instantly refresh_
_- Fixed error fetch from Auth (From login page when you denied)_
_- Refreshing page keeps you in the same page_

Leaderboard page:
_- Sort on Elo Rating, if players do have the same score the most recently registered player will be on top_

Home page: 
_- When not logged, content is responsive_

### CODE REVIEW

File: srcs/client/src/Pages/Home/home.tsx [hunk 1]
```diff
@@ -1,73 +1,162 @@
...
 
-  useEffect(() => {
-    localStorage.getItem("pathIsFree");
-    console.log("test");
-  });
+/*
+** First function to Auth the user from our app -> 42Auth -> back -> front 
+*/
+const getUser = async () => {
+  await  fetch("http://127.0.0.1:4200/auth/success", {
```
Comment (twagger) [inline srcs/client/src/Pages/Home/home.tsx:23]: Ici on passe directement par le port mappe sur la machine hote. Ca ne marchera pas en production, il faudra passer par la redirection nginx et donc mettre notre domain name si on en a un ou sinon taper directement sur le front (127.0.0.1:8080 par exemple). Il faudra commencer a prevoir ca sinon ca va etre complique d'ici 2 semaines

File: srcs/client/src/index.tsx [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
 import App from './App';
 
-import { BrowserRouter } from 'react-router-dom';
-
+/*
+** Root of the project 
+*/
 const root = ReactDOM.createRoot(
-  document.getElementById('root') as HTMLElement
+  document.getElementById('root') as HTMLElement as HTMLElement
```
Comment (twagger) [inline srcs/client/src/index.tsx:9]: Doublon ?

File: srcs/server/src/auth/filters/ftAuthException.filter.ts [hunk 3]
```diff
@@ -13,6 +13,6 @@ export class FtExceptionFilter implements ExceptionFilter {
     const context = host.switchToHttp();
     const response = context.getResponse<Response>();
 
-    response.redirect('http://127.0.0.1:3042/home');
+    response.redirect(process.env.LOGIN_PAGE);
```
Comment (twagger) [inline srcs/server/src/auth/filters/ftAuthException.filter.ts:16]: :+1:

**Review by twagger — APPROVED** [review]
Ok for me

**Review by jnougaro — APPROVED** [review]
It seems fine to me :)

**Pull request merged**